### DISCUSSION TITLE
reviewers should be added also in case of automerge

### DISCUSSION THREAD

**Discussion by zplan** [renovatebot/renovate#18039] [Suggest an Idea]
Scenario:
- running renovate against several azure repos with pr & automerge
- the repos have a pr build policy, so only green builds can be auto-merged
- if pr build is green it is automerged (autocompleted)
- if pr build is red, pr stays open and must be fixed

Issue:
- I add an (optional) reviewer for the pr in the renovate config file
- the reviewer is not added because the (debug) log prints: "skip adding reviewers because of automerge=true"

--> But I would like  to have an optional reviewer added because
- I want to get a pr notification
- in case the build is red, I want to get another pr notification
- in case the build is red, I want to be the reviewer of the fix on this branch

Conclusion:
It makes only sense to skip the mandatory reviewer if automerge=true but it should add optional reviewers anyway.

**Comment by rarkins** [comment]
Does Renovate understand mandatory vs optional reviewers for Azure already? Or is that a new concept too?

**Reply by zplan** [reply]
I can add:
`reviewers: ['abc@abc.abc'],`
This adds an optional reviewer in azure, which is fine in my case.

> mandatory vs optional reviewers

Not sure about the current state, I think discussion is still  ongoing in another discussion thread: https://github.com/renovatebot/renovate/discussions/17659

**Comment by viceice** [comment]
the reviewer should be added even on automerge after 24h of non green or state

**Reply by zplan** [reply]
> the reviewer should be added even on automerge after 24h of non green or state

Thanks, can the time be configured?

**Reply by viceice** [reply]
yes

https://docs.renovatebot.com/configuration-options/#prnotpendinghours

**Reply by zplan** [reply]
Thanks, regarding this scenario  I have another question:
I use an azure pipeline to run renovate with a docker image like: https://docs.renovatebot.com/examples/self-hosting/#docker
After the `docker run` command has finished the container ist stopped. So in this scenario all the time related properties are not working, I would need to have a renovate instance always running, right?

**Reply by viceice** [reply]
no, you should run the pipeline hourly, the oss renovate is a cli program only. so you need to run it as often as you like to have updates.

use schedule option to limit specific updates to specific the ranges
